# <center>House price prediction with machine learning models 🤖</center>

![house](image.png)

## Contents

- [Getting the data](#getting-the-data)
- [Data analysis](#data-analysis)
- [Data cleaning](#data-cleaning)
- [Data visualization](#data-visualization)
- [Setting up the models](#setting-up-the-models)
- [Training the models](#training-the-models)

## Data description 📝

- `id` - Unique ID for every individual row entry  
- `date` - Date the house was sold  
- `price` - Price of the house  
- `bedrooms` - Number of bedrooms  
- `bathrooms` - Number of bathrooms  
- `sqft_living` - Square footage of the house  
- `sqft_lot` - Square footage of the lot  
- `floors` - Number of floors  
- `waterfront` - Whether the house has a view to a waterfront  
- `view` - How good the view of the property is  
- `condition` - Condition of the house  
- `grade` - Grade given to the house based on the overall construction and design  
- `sqft_above` - Square footage of house apart from the basement  
- `sqft_basement` - Square footage of the basement  
- `yr_built` - Year the house was built  
- `yr_renovated` - Year the house was renovated  
- `zipcode` - Zipcode of the house  
- `lat` - Latitude coordinate  
- `long` - Longitude coordinate  
- `sqft_living15` - Living room area in 2015(implies-- some renovations) This might or - might not have affected the lotsize area  
- `sqft_lot15` - LotSize area in 2015(implies-- some renovations)  

<hr>

In [35]:
import numpy as np
import pandas as pd
import seaborn as sn
import matplotlib.pyplot as plt

## Getting the data

In [36]:
# Create a dataframe from the csv file
df = pd.read_csv('kc_house_data.csv')

In [37]:
# Shape of the dataframe
df.shape

(21613, 21)

<hr>

## Data exploration

In [38]:
# Display the first 5 rows of the dataframe
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

In [39]:
# Display the last 5 rows of the dataframe
df.tail()

id             date     price  bedrooms  bathrooms  \
21608   263000018  20140521T000000  360000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[5 rows x 21 columns]

In [40]:
# Sample random 10 rows from the dataframe
df.sample(10)

id             date      price  bedrooms  bathrooms  \
21082  8562900430  20140718T000000   800000.0         4       2.50   
10426  6746700565  20141023T000000   447000.0         2       1.00   
9693   8644300200  20140605T000000   555000.0         4       2.75   
12279  2313900810  20150402T000000   610000.0         4       2.00   
17248   725069102  20150330T000000   650000.0         3       2.25   
16567  7603100095  20141110T000000  1260000.0         3       3.00   
2079   1336800160  20140605T000000   875000.0         5       2.50   
584    5419800510  20141117T000000   268500.0         4       1.75   
19849  7708210070  20140617T000000   535000.0         4       2.75   
12868  4014400237  20140523T000000   132500.0         3       1.00   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
21082         3691     11088     2.0           0     1  ...      8   
10426          850      2700     1.0           0     0  ...      6   
9693          2020     10720     1.0           0     0  ...      8   
12279         2220      5821     1.5           0     0  ...      7   
17248         2180     60112     2.0           0     0  ...      8   
16567         3230      8625     2.0           0     3  ...     10   
2079          2920      5568     2.0           0     0  ...      8   
584           1420      7500     1.0           0     0  ...      7   
19849         3070      7201     2.0           0     0  ...      9   
12868         1080     10500     1.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
21082        3691              0      2013             0    98074  47.6122   
10426         850              0      1924             0    98105  47.6684   
9693         1420            600      1976             0    98052  47.6373   
12279        1380            840      1916             0    98116  47.5723   
17248        2180              0      1976             0    98053  47.6723   
16567        2220           1010      1998             0    98116  47.5620   
2079         2320            600      1906             0    98112  47.6265   
584          1080            340      1981             0    98031  47.4025   
19849        3070              0      2006             0    98059  47.4897   
12868        1080              0      1967             0    98001  47.3200   

          long  sqft_living15  sqft_lot15  
21082 -122.059           3190       11270  
10426 -122.316           1630        3000  
9693  -122.104           2190       10164  
12279 -122.382           1850        5000  
17248 -122.082           2060      120225  
16567 -122.404           2330        6022  
2079  -122.312           2970        5568  
584   -122.176           1500        7260  
19849 -122.147           2880        8364  
12868 -122.278           1200        9607  

[10 rows x 21 columns]

In [41]:
# Display the columns of the dataframe
df.columns

Index(['id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'],
      dtype='object')

In [42]:
# Display info and the datatypes of the columns
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

### Data shows the following:

- There are 21,613 rows or entries in the dataset
- There are 21 columns or features in the dataset
- There are different data types in the dataset, including `int64`, `float64`, and `object`
- There are no missing values in the dataset
- There are 15,458 unique values in the `id` column
- Memory usage of the dataset is 3.5+ MB

## Data cleaning

## Data visualization

Contenido de la sección "Data visualization"

## Setting up the models

Contenido de la sección "Setting up the models"

## Training the models

Contenido de la sección "Training the models"